src/lib/y2packager/clients/inst_productsources.rb

```diff
@@ -358,17 +359,11 @@ def NetworkRunning
     def NormalizeURL(url_string)
       return url_string if url_string.nil? || url_string == ""
 
-      if Builtins.regexpmatch(url_string, "/+$")
-        url_string = Builtins.regexpsub(url_string, "(.*)/+$", "\\1")
-      end
-
-      # URL is escaped
-      if Builtins.regexpmatch(url_string, "%")
-        # unescape it
-        url_string = URL.UnEscapeString(url_string, URL.transform_map_filename)
-      end
+      # removing all slashes at the end of the url
+      url_string.sub!(/(\/)+$/, "")
 
-      url_string
+      # unescape it
+      URI.unescape(url_string)
```

Comment:
Just a theoretical question, what if the slash at the end is escaped? I think first we should unescape and then remove the trailing slashes...